src/Raven.Server/Utils/CpuUsageExtensionPoint.cs

```diff
@@ -0,0 +1,263 @@
+        {
+            get
+            {
+                if (IsDisposed && _logger.IsOperationsEnabled)
+                {
+                    _logger.Operations($"Try to get data from {nameof(CpuUsageExtensionPoint)} which was disposed");
+                }
+                return _data;
+            }
+        }
+
+        public bool IsDisposed { get; private set; }
+
+
+        public CpuUsageExtensionPoint(
+            JsonContextPool contextPool,
+            string exec,
+            string args,
+            Logger logger,
+            NotificationCenter.NotificationCenter notificationCenter)
+        {
+            _contextPool = contextPool;
+            _logger = logger;
+            _notificationCenter = notificationCenter;
+            _startInfo = new ProcessStartInfo
+            {
+                FileName = exec,
+                Arguments = args,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            };
+
+            InitNewProcessHandle();
+        }
+
+        public void Start()
+        {
+            try
+            {
+                _process.Start();
+            }
+            catch (Exception e)
+            {
+                NotifyWarning("Could not start cpu usage extension point process", e);
+
+                Dispose();
+            }
+
+            var _ = ReadProcess(); // explicitly starting async task without waiting for it
+            _ = ReadErrors();
+        }
+
+        private async Task ReadErrors()
+        {
+            var errors = await _process.StandardError.ReadLineAsync();
+            if (errors == null)
+            {
+                return;
+            }
+            try
+            {
+                if (_process.HasExited == false)
+                {
+                    _process.Kill();
+                }
+            }
+            catch (Exception)
+            {
+                // When the process terminating, killed or exited
+            }
+            errors = errors + await _process.StandardError.ReadLineAsync();
+            NotifyWarning($"Extension point process send an error: {errors}");
+        }
+
+        public void Dispose()
+        {
+            IsDisposed = true;
+            _data = BadData;
+            using (_process)
+            {
+                try
+                {
+                    if (_process.HasExited == false)
+                    {
+                        _process.Kill();
+                    }
+                }
+                catch (Exception)
+                {
+                    // When the process terminating, killed or exited
+                }
+
+                _process = null;
+            }
+        }
+
+
+        private async Task ReadProcess()
+        {
+            while (IsDisposed == false)
+            {
+                var nextLine = _process.StandardOutput.ReadLineAsync();
```

Comment:
We should Try-Catch here, we don't want an unhandled failed task, we could for an instance have a temporary out of memory problem here, that will just kill the task, without us knowing/monitoring it.